spec/controllers/announcements_controller_spec.rb

```diff
@@ -1,182 +1,130 @@
+      it_behaves_like 'redirected request'
     end
     context 'POST create' do
-      before do
-        post :create, announcement: @attributes
-      end
-      it_behaves_like 'access denied'
+      before { post :create, announcement: { id: 1 } }
+      it_behaves_like 'redirected request'
     end
     context 'PUT update' do
-      before do
-        put :update, id: @announcement
-      end
-      it_behaves_like 'access denied'
+      before { put :update, id: 1 }
+      it_behaves_like 'redirected request'
     end
     context 'DELETE destroy' do
-      before do
-        delete :destroy, id: @announcement
-      end
-      it_behaves_like 'access denied'
+      before { delete :destroy, id: 1 }
+      it_behaves_like 'redirected request'
     end
   end
-  context 'GET hide as' do
-    shared_examples 'can hide announcement' do
+
+  describe 'GET hide as' do
+    shared_examples 'can hide announcement' do |user_type|
+      let!(:announcement) { AnnouncementMock.new(traits: [:findable]) }
       before do
-        @announcement = FactoryGirl.create(:announcement)
+        mock_user_sign_in(UserMock.new(user_type))
         request.env['HTTP_REFERER'] = 'where_i_came_from'
-        get :hide, id: @announcement
+        get :hide, id: announcement.id
       end
-      it 'should set some cookie values' do
+      it 'sets some cookie values' do
         name = 'hidden_announcement_ids'
         jar = request.cookie_jar
-        jar.signed[name] = [@announcement[:id].to_s]
+        jar.signed[name] = [announcement.id.to_s]
         expect(response.cookies[name]).to eq(jar[name])
       end
     end
-    context 'superuser' do
-      before do
-        sign_in FactoryGirl.create(:superuser)
-      end
-      it_behaves_like 'can hide announcement'
-    end
-    context 'admin' do
-      before do
-        sign_in FactoryGirl.create(:admin)
-      end
-      it_behaves_like 'can hide announcement'
-    end
-    context 'patron' do
-      before do
-        sign_in FactoryGirl.create(:user)
-      end
-      it_behaves_like 'can hide announcement'
-    end
-    context 'checkout person' do
-      before do
-        sign_in FactoryGirl.create(:checkout_person)
-      end
-      it_behaves_like 'can hide announcement'
-    end
-    context 'guest' do
-      before do
-        sign_in FactoryGirl.create(:guest)
-      end
-      it_behaves_like 'can hide announcement'
-    end
-    context 'banned user' do
-      before do
-        sign_in FactoryGirl.create(:banned)
-      end
-      it_behaves_like 'can hide announcement'
-    end
+    # ROLES = [:superuser, :admin, :checkout_person, :user, :guest, :banned]
+    # ROLES.each { |r| it_behaves_like 'can hide announcement', r }
+    PASSING_ROLES = [:superuser, :admin].freeze
+    PASSING_ROLES.each { |r| it_behaves_like 'can hide announcement', r }
+    FAILING_ROLES = [:checkout_person, :user, :guest, :banned].freeze
+    FAILING_ROLES.each { |r| it_behaves_like 'can hide announcement', r }
```

Comment:
These fail due to a permissions error I have yet to fully figure out. This gets thrown to the `rescue_from CanCan::AccessDenied` block in `ApplicationController`, but when I put a `binding.pry` there I discovered that `current_ability.can? :hide, Announcement` returns true...